core/src/commonMain/kotlin/com/xebia/functional/xef/auto/AIRuntime.kt

```diff
@@ -0,0 +1,32 @@
+package com.xebia.functional.xef.auto
+
+import com.xebia.functional.xef.embeddings.OpenAIEmbeddings
+import com.xebia.functional.xef.env.OpenAIConfig
+import com.xebia.functional.xef.llm.openai.KtorOpenAIClient
+import com.xebia.functional.xef.llm.openai.LLMModel
+import com.xebia.functional.xef.vectorstores.LocalVectorStore
+import kotlin.jvm.JvmStatic
+import kotlin.time.ExperimentalTime
+
+data class AIRuntime<A>(val runtime: suspend (block: AI<A>) -> A) {
+  companion object {
+    @OptIn(ExperimentalTime::class)
+    @JvmStatic
+    fun <A> openAI(): AIRuntime<A> = AIRuntime { block ->
+      val openAIConfig = OpenAIConfig()
+      KtorOpenAIClient(openAIConfig).use { openAiClient ->
+        val embeddings = OpenAIEmbeddings(openAIConfig, openAiClient)
+        val vectorStore = LocalVectorStore(embeddings)
+        val scope =
+          AIScope(
+            defaultModel = LLMModel.GPT_3_5_TURBO_16K,
+            defaultSerializationModel = LLMModel.GPT_3_5_TURBO_FUNCTIONS,
```

Comment:
Are the serialization models generic to all the platforms? Or are they only specific to OpenAI?

Reply:
No, we expect that for local models we will have to implement the function interface similar to OpenAI, desugaring the request to whatever serialization mechanism we can use with a local model that already specs json from a schema.
We also plan to split the types of the features of the client so they are more in the lines of Audio, Video, Text, Chat, etc.. and these are typed so you are unable to run a `prompt` or `image` in models where it does not make sense. We still need to implement all this.